### ISSUE TITLE
Someday: programmatically add new ctfe free standing functions to replicate eg methods and other functions that take vector arguments but with D types instead 

### ISSUE THREAD

**Issue by Laeeth** [atilaneves/dpp#128]
So they just make a vector, map, string etc from the D side argument, call the cpp, and translate the return if needed

**Comment by atilaneves** [comment]
If I'm getting this right, then it would be a translation layer similar to excel-d and autowrap:python?

**Comment by Laeeth** [comment]
Yes I guess so although it should be much less work if you don't care about perfection provided it's correct.
